### PR TITLE
feat(WEG-143): Desktop interaction for multiple points in same coordinates

### PR DESCRIPTION
**Pull request by vogelino**
This PR adds a hover tooltip on desktop to make the experience quicker by removing the step in between of having to click on the slider elements again. It also adds a expandable "spider web" or "branching" effect on markers with multiple facilities in the same geolocation. The mobile experience remains untouched.

|Desktop|Mobile|
|--|--|
|<video src="https://user-images.githubusercontent.com/2759340/205599651-09f198cd-b5fb-40a6-af0d-f8870092550b.mov" />|<video src="https://user-images.githubusercontent.com/2759340/205599681-99a46245-2e55-4bb2-b409-0cd2d59c1173.mov" />|

### CODE REVIEW

File: src/components/Map/index.tsx [hunk 1]
```diff
@@ -44,6 +52,7 @@ export const transitionProps = {
   transitionEasing: easeInOutQuad,
 }
 
+const MAX_ZOOM = 17
```
Comment (dnsos) [inline src/components/Map/index.tsx:55]: A few lines below `MAP_CONFIG.maxZoom` is defined. Perhaps we should unify the values, unless there is a reason for why they are different?

File: src/components/Map/mapUtil.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  maxZoom = 17
+): void {
+  map.easeTo({
+    center: coordinates,
+    zoom: Math.min(maxZoom, map.getZoom() + zoomIncrease),
+  })
+}
+
+export function setCursor(cursor = 'grab'): void {
+  const container$ = document.querySelector<HTMLDivElement>(
```
Comment (dnsos) [inline src/components/Map/mapUtil.ts:67]: Hmm, I've seen that use of $ before my I can't recall what it means. What is indicated with it?

File: src/lib/hooks/useDistanceToUser.ts [hunk 3]
```diff
@@ -11,21 +11,29 @@ interface UseDistanceToUserReturnType {
   getDistanceToUser: (pointA: PointType) => number | undefined
 }
 
+export function getDistanceBetweenTwoPoints(
```
Comment (dnsos) [inline src/lib/hooks/useDistanceToUser.ts:14]: I wonder if that function could be refactored a bit: as the name suggests any two points could be put in. How about renaming the variables within accordingly and getting rid of references to user and facility? That could be taken one step further by moving the function somewhere more general-map-related such as `mapUtil.ts`?

**Review by dnsos — APPROVED** [review]
Big props for this work @vogelino! It's a great feature and while a lot of complexity is added you managed to keep it as clean as possible, in my impression. Also great job for typing and incorporating that Spiderifier! 

I've got a few stylistic comments that would be nice-to-be-addressed, but nothing critical. Well done!

File: src/components/Map/mapUtil.ts (hunk 2 above)
Comment (vogelino) [reply src/components/Map/mapUtil.ts:67]: It indicates a DOM node is saved into this variable. But I'm not sure we are following this convention consistently... :/

File: src/lib/hooks/useDistanceToUser.ts (hunk 3 above)
Comment (vogelino) [reply src/lib/hooks/useDistanceToUser.ts:14]: I like your suggestion! I'll do this!

**Pull request merged**